src/lib/transport.js

```diff
@@ -155,134 +105,146 @@ async function listenAll (factory, {
   *
   * Note return values are only for testing
   */
-async function _autoFulfillCondition ({
-  transfer,
+async function handleData ({
   plugin,
   receiverSecret,
   allowOverPayment,
-  minFulfillRetryWait,
-  maxFulfillRetryWait,
   callback: reviewFunction
-}) {
-  const account = plugin.getAccount()
+}, packet) {
+  const account = await ILDCP.getAccount(plugin)
 
-  // TODO: should this just be included in this function?
-  const err = await _validateOrRejectTransfer({
-    plugin,
-    transfer,
-    allowOverPayment,
-    receiverSecret
-  })
+  try {
+    const transfer = IlpPacket.deserializeIlpPrepare(packet)
+    const parsed = IlpPacket.deserializeIlpPayment(transfer.data)
```

Comment:
Putting the old packet type inside the new packet type is a pretty strange hack to make this change less breaking, so it would definitely be good to include a comment explaining this